search/search-client-elasticsearch/src/main/java/io/camunda/search/es/transformers/aggregator/ProcessDefinitionFlowNodeStatisticsAggregationTransformer.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.search.es.transformers.aggregator;
+
+import static io.camunda.search.query.ProcessDefinitionFlowNodeStatisticsQuery.*;
+
+import co.elastic.clients.elasticsearch.core.SearchResponse;
+import io.camunda.search.aggregation.ProcessDefinitionFlowNodeStatisticsAggregation;
+import io.camunda.search.clients.transformers.SearchTransfomer;
+import io.camunda.search.entities.ProcessDefinitionFlowNodeStatisticsEntity;
+
+public class ProcessDefinitionFlowNodeStatisticsAggregationTransformer
+    implements SearchTransfomer<SearchResponse<?>, ProcessDefinitionFlowNodeStatisticsAggregation> {
+
+  @Override
+  public ProcessDefinitionFlowNodeStatisticsAggregation apply(final SearchResponse<?> value) {
```

Comment:
@danielkelemen, can't this be moved one layer above, i.e., to `the search-client-query-transformer` to avoid having this logic in two places (ES and OS)?

Reply:
@romansmirnov, we would still need to transform the ES and OS aggregation results into one intermediate result, like a hash map for instance. And then we could have one statistics transformer in the `client-query-transformer` which transforms that into the final result.
I could give this a try, it would probably make sense in the future. I'd probably do this as a follow-up.

Reply:
Thanks for clarifying. I think it would be worth trying it. Because that way the "transformers" on this level remain business-logic agnostic, and the business-logic can be implement only once on top of it.